cogs/commands/flip.py

```diff
@@ -22,13 +22,30 @@ async def flip(self, ctx: Context, *, args: str = ""):
         args = await clean_content().convert(ctx, args)
         args = shlex.split(args)
         display_name = get_name_string(ctx.message)
+        weighted = False
 
         if len(args) == 1:
             await ctx.send(f"I can't flip just one item {display_name}! :confused:")
-        else:
-            options = ["Heads", "Tails"] if not args else args
 
-            await ctx.send(f'{display_name}: {random.choice(options).lstrip("@")}')
+        else:
+            if "weights" in [arg.lower() for arg in args]:
+                weight_index = args.index("weights")
+                if len(args[:weight_index]) == len(args[: weight_index + 1 :]):
+                    options = args[:weight_index]
+                    option_weights = [
+                        int(weight) for weight in args[weight_index + 1 :]
+                    ]
+                    weighted = True
+                else:
```

Comment:
```suggestion
                weighted = True
                options = args[:weight_index]
                option_weights = [int(weight) for weight in args[weight_index + 1 :]]
                if len(options) != len(option_weights):
```
I think moving `options` and `options_weights` outside the if statement make the control flow a little clearer. Also there is a stray colon inserted on the RHS of the original condition.